src/compiler/moduleNameResolver.ts

```diff
@@ -96,9 +96,14 @@ namespace ts {
         };
     }
 
+    function tryParsePatterns(paths: MapLike<string[]>): (string | Pattern)[] {
+        return mapDefined(getOwnKeys(paths), path => tryParsePattern(path));
+    }
+
     interface ModuleResolutionState {
         host: ModuleResolutionHost;
         compilerOptions: CompilerOptions;
+        compilerPathPatterns?: (string | Pattern)[];

```

Comment:
This helps, but patterns are still being parsed many times per tsconfig.  @sheetalkamat is there an upstream cache I should be using instead?